### PR TITLE
Implemented timezone support, #485

### PR DESCRIPTION
**Pull request by e-vandenberg**
Added feature from issue #485 to allow custom timezones with datetime.

Below is an example of usage:

```python
>>> d = Datetime()
>>> d.datetime(timezone="Europe/Paris")
datetime.datetime(2003, 12, 18, 4, 53, 48,  tzinfo=<DstTzInfo 'Europe/Paris' CEST+2:00:00 DST>)
```

This enhancement requires pytz.

### CODE REVIEW

**Comment by lk-geimfari** [comment]
We can't add dependencies only for this little feature, os i think that must be optional feature, I suggest something like this:

```python

    def datetime(self, humanized: bool = False, timezone: str = '', **kwargs) -> DateTime:
        """Generate random datetime.
        :param humanized: Readable representation.
        :param timezone: Custom timezone.
        :param kwargs: Keyword arguments (start, end).
        :return: Datetime.
        :rtype: datetime.datetime
        :Example:
            March, 24 2002
        """
        fmt = '%Y-%m-%d %H:%M:%S'
        dt_str = '{date} {time}'.format(
            date=self.date(fmt='%Y-%m-%d', **kwargs),
            time=self.time(),
        )

        dt = datetime.strptime(dt_str, fmt)

        if humanized:
            return dt.strftime('%B, %d %Y')

        if timezone != '':
            try:
                import pytz
                timezone = pytz.timezone(timezone)
                return  timezone.localize(dt)
            except ImportError:
              raise ImportError('Timezones supported only with pytz')
        return dt
```

@sobolevn What do you think?

**Comment by sobolevn** [comment]
Agreed about not adding a direct dependency.

I guess we can create some kind of `compat.py` file with the following contents:

```python
try:
    import pytz
except ImportError:
   pytz = None
```

And then use it like so:

```python
from mimesis.compat import pytz

def datetime(self, humanized: bool = False, timezone: str = '', **kwargs) -> DateTime:
       ...

        if timezone:
            if not pytz:
                raise ImportError('Timezones are supported only with pytz')
           ...
        return dt
```

**Comment by lk-geimfari** [comment]
@sobolevn Totally agree. @e-vandenberg Please, look at the comment above.

File: mimesis/providers/date.py [hunk 1]
```diff
@@ -176,10 +177,11 @@ def gmt_offset(self) -> str:
         """
         return self.random.choice(GMT_OFFSETS)
 
-    def datetime(self, humanized: bool = False, **kwargs) -> DateTime:
+    def datetime(self, humanized: bool = False, timezone: str = '', **kwargs) -> DateTime:
         """Generate random datetime.
 
         :param humanized: Readable representation.
+        :param timezone: Custom timezone.
```
Comment (sobolevn) [inline mimesis/providers/date.py]: We need to add a note about `pytz` here. We should also mention that installing `pytz` is required to use it.

**Comment by lk-geimfari** [comment]
@e-vandenberg Also, please don't forget about tests. You can add `pytz` to dev dependencies and test this feature.

**Comment by e-vandenberg** [comment]
@lk-geimfari @sobolevn Thank you both very much for the feedback and support. Is there anything else I should add to this PR?

File: mimesis/providers/date.py [hunk 2]
```diff
@@ -198,6 +202,13 @@ def datetime(self, humanized: bool = False, **kwargs) -> DateTime:
         if humanized:
             return dt.strftime('%B, %d %Y')
 
+        if timezone != '':
+            if not pytz:
+                raise ImportError('Timezones are supported only with pytz')
+            else:
```
Comment (sobolevn) [inline mimesis/providers/date.py]: We do not need `else` here.

File: tests/test_providers/test_datetime.py [hunk 3]
```diff
@@ -113,6 +114,13 @@ def test_datetime(self, _datetime, start, end, humanized, _type):
             year = int(dt.split(' ')[2])
             assert year == 2018
 
+        if humanized == True:
```
Comment (sobolevn) [inline tests/test_providers/test_datetime.py]: Unneeded condition.

**Comment by sobolevn** [comment]
Please, also, pay attention to the travis build. Some checks are failing.

File: tests/test_providers/test_datetime.py (hunk 3 above)
Comment (e-vandenberg) [reply tests/test_providers/test_datetime.py]: If humanized is passed as `True` there will be no `tzinfo` attribute and the proceeding lines will result in an error. This `if ... pass` makes sure `tzinfo` isn't checked when the datetime is humanized. 

Maybe I am missing something, but I can't seem to find a better way to do this. Did you have anything in mind?

**Comment by e-vandenberg** [comment]
I have fixed the formatting issues, and most of the bugs, however, there are two remaining errors (as seen in both the Travis CI and AppVeyor tests) that I have not been able to resolve. 

- The first is in `test_datetime.py` and shows no error message, however it comes up as a failure. I am not sure what to make of this. Below is he test result.
```python
def test_datetime(self, _datetime, start, end, humanized, timezone, _type):
        dt = _datetime.datetime(start=start, end=end,
>                                               humanized=humanized, timezone=timezone)
```
- The second one is an `ImportError` that is called in the program. I believe to fix this, `pytz` needs to be added properly to the `Pipfile` and `Pipfile.lock`, however I am having trouble adding to the `Pipfile.lock`

Any help with these small issues would be greatly appreciated.

**Comment by sobolevn** [comment]
@e-vandenberg try to `pipenv install --dev pytz`. This will add `pytz` to both `Pipefile` and `Pipfile.lock`. 

I have also opened a new issue about our documentation missing "Adding dependencies" section: https://github.com/lk-geimfari/mimesis/issues/498

**Comment by e-vandenberg** [comment]
Travis CI checks passed, some AppVeyor checks passed. Do all checks need to pass?

**Comment by lk-geimfari** [comment]
@e-vandenberg Can you please recreate PR?

**Comment by lk-geimfari** [comment]
I have add it by myself in #539

**Pull request closed without merging**